BD-22: Add My Activities tab

deb1990 opened the pull request:
## Overview
As part of this PR, the Activity Feed is added to the My Activities tab.

## After
![after](https://user-images.githubusercontent.com/5058867/72068621-7f341d00-330b-11ea-91cc-9196117b05c0.gif)


## Technical Details
1. In `DashboardTabs.php` added Angular Loader so that angular directives can be used.
2. In `OptionValues.php` added `filter` field. As it is used to fetch the Activity Status's with `Incomplete category` https://github.com/compucorp/uk.co.compucorp.civicase/pull/325/files#diff-ffa4750072bb1f58bec7b566fda6e147R25
3. In `activity-filters-affix.directive.js`, a new function `getTabsElement` is created to support affix when inside crm dashboard.
4. `crm-dashboard-tabset-affix.directive.js` is created to affix the CRM Dashboard tabs.
5. `civicaseMyActivitiesTab` directive is created which is responsible for displaying the Activity Feed with pre selected filters.
6. `activity-feed.directive.js` is refactored. Previously `$scope.filters` and `$scope.params.filters` were merged just before making the API call, but because of this, if `@involvingContact` param is set through `$scope.params.filters`, then it was not being respected, because the check happened before merging 2 objects
https://github.com/compucorp/uk.co.compucorp.civicase/pull/325/files?file-filters%5B%5D=.css&file-filters%5B%5D=.html&file-filters%5B%5D=.js&file-filters%5B%5D=.scss&file-filters%5B%5D=.tpl#diff-a2691041d879770b82d5927fa9d0f10dR391

7. Also now `ActivityFeed` supports filters with `$` prefix. This can be used when we want to directly apply the filter without going through the existing logic https://github.com/compucorp/uk.co.compucorp.civicase/pull/325/files?file-filters%5B%5D=.css&file-filters%5B%5D=.html&file-filters%5B%5D=.js&file-filters%5B%5D=.scss&file-filters%5B%5D=.tpl#diff-a2691041d879770b82d5927fa9d0f10dR431

## Comments
Because of time constraints I could not write unit tests.

reneolivo commented on ang/civicase/myactivities/directives/crm-dashboard-tabset-affix.directive.js:
```diff
@@ -0,0 +1,61 @@
+(function (angular, $) {
```
I think this should be inside a `ang/civicase/crm-dashboard` folder.

reneolivo replied:
But tbh, this should be a generic implementation if possible. Not sure how long it would take, but I see that we repeat the same pattern.

I think we can have something like:

```
<div civicase-affix-after="civicase__crm-dashboard">...</div>
```

I'm not sure about all these calculations, but isn't it enough to know only how far from the top the main container element is?

deb1990 replied:
I agree, and I know I have been ignoring this particular bit, but honestly this would take much longer time if I start refactoring all directives now. then we wont be able to finish it by tomorrow, as the testing scope will increase a lot too.

reneolivo commented on ang/civicase/myactivities/directives/my-activities-tab.directive.js:
```diff
@@ -0,0 +1,41 @@
+     */
+    function getIncompleteActivityStatusIDs () {
+      return _.filter(ActivityStatus.getAll(), function (status) {
+        return status.filter === '0';
```
```js
// somewhere at the top:
var INCOMPLETE_ACTIVITY_STATUS = '0';

// ...
return status.filter === INCOMPLETE_ACTIVITY_STATUS;
```

jamienovick commented:
@deb1990 - I see a scroll bar has appears on the activities feed - is that something we can fix?

deb1990 commented:
> @deb1990 - I see a scroll bar has appears on the activities feed - is that something we can fix?

The scrollbar is intentional and appears on other pages too. This was done so that the Month Navigation Always stays on the right side of the Activity List even when we scroll the list.

Do you want me to remove it?

jamienovick commented:
@deb1990 when was it added? Can I see the discussion. Yes I don't think we should have a scroll bar; this is a huge change to the UX... !!

deb1990 commented:
**Fixed the shadows**
Before
![2020-01-15 at 12 24 PM](https://user-images.githubusercontent.com/5058867/72411894-052fe800-3792-11ea-99b9-9fc006274c19.jpg)

After
![2020-01-15 at 12 23 PM](https://user-images.githubusercontent.com/5058867/72411800-cb5ee180-3791-11ea-9ba6-55f472921048.jpg)

Technical Details
1. Added Z-index in `_crm-dashboard.scss`.

**Fixed the affix for Activity Filter Affix**
Before
![before](https://user-images.githubusercontent.com/5058867/72411995-43c5a280-3792-11ea-9434-e749a61cf11c.gif)

After
![after](https://user-images.githubusercontent.com/5058867/72412048-635ccb00-3792-11ea-8b55-135400a553a4.gif)

Technical Details
1. In `activity-filters-affix.directive.js` replaced `#toolbar` with `#civicrm-menu`, to fix the affix behaviour.

Also fixed broken unit tests from previous commits.

jamienovick commented:
♥️

swastikpareek commented on CRM/Civicase/Page/DashboardTabs.php:
```diff
@@ -9,6 +11,14 @@ class CRM_Civicase_Page_DashboardTabs extends CRM_Contact_Page_DashBoard {
    * Adds the CRM dashboard's CSS file and Dashboard page content.
    */
   public function run() {
+    $loader = new AngularLoader();
```
Shouldn't var initialisation have an empty line after  it ?

swastikpareek commented on ang/test/civicase/case/details/directives/case-details.directive.spec.js:
```diff
@@ -255,6 +255,9 @@
     });
 
+    /**
+     *
```
Please add doc block
